vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp

```diff
@@ -1,23 +1,205 @@
+    
+    
+    
+    //------- edit title of Sequence starts --------
+    
+    $("#submitTitle").hide();
+    $("#cancelEditTitle").hide();
+    
+    $("#editTitle").click(function() {
+    	// Using data attribute
+    	var sequenceTitle = $("#titleName").data('value')  // gets value without using trim function
+        $('<div class="col-4" id = "editSequenceTitle"><input id="newTitle" class="form-control" type="text"></div>').insertAfter( "#title" );
+    	$("#title").val(sequenceTitle)
+    	$('#title').text('Sequence: ')
+        $("#newTitle").val(sequenceTitle)
+        $("#editTitle").hide()
+        $("#submitTitle").show()
+        $("#cancelEditTitle").show()
+    });
+    	
+    	
+    	
+    // ------ Submit title Sequence -------
+    	
+    $("#submitTitle").click(function() {
+        var formData = new FormData();
+        formData.append('title', $("#newTitle").val());
+        $.ajax({
+            url: "<c:url value="/staff/module/${module.id}/sequence/${sequence.id}/edit/title?${_csrf.parameterName}=${_csrf.token}" />",
+            type: 'POST',
+            cache       : false,
+            contentType : false,
+            processData : false,
+            data: formData,
+            enctype: 'multipart/form-data',
+            success: function(data) {
+                // replace text box with new description
+                $("#submitTitle").hide()
+                $("#cancelEditTitle").hide();
+                $("#editTitle").show()
+                var val = $("#newTitle").val();
+                $("#editSequenceTitle").remove()
+                $("#title").text("Sequence: " + val)
+                $("#titleName").data('value', val)  // sets value
+            },
+            error: function(data) {
+                var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+                $('.error').append(alert);
+            }
+        });
+                
+      });
+    
+    
+   // ------  Edit Title Sequence ------
+   
+   $("#cancelEditTitle").click(function(){
+        $("#submitTitle").hide()
+        $("#editTitle").show()
+        $("#cancelEditTitle").hide()
+        var sequenceTitle = $("#titleName").data('value')  // gets value
+        $("#editSequenceTitle").remove()
+        $("#title").text("Sequence: "+sequenceTitle)
+  
+    });
+   // ------- Edit/Save Sequence Title ends --------
+});
+
+
+
+</script>
+
+
+<!-- title -->
+<div class="row align-items-center">
+	<h2 id="title" style="margin-bottom: 0%; margin-left: 1%;" data-value="${sequence.name}">
+		Sequence: ${sequence.name}
+		</h2>
+		<p id="titleName" style="margin-top: .5rem; font-size:35px; margin-bottom: .5rem;" data-value="${sequence.name}"></p>
```

Comment:
why is this here?

Reply:
I used the p tag, since h2 tag cannot have val attribute. So in order to retrieve the Sequence name, I used p tag and got the value using data-attribute

Reply:
data attributes should be valid on any tag as far as I know